crates/fuzzing/src/oracles.rs

```diff
@@ -645,3 +654,32 @@ pub fn differential_wasmi_execution(wasm: &[u8], config: &crate::generators::Con
 
     Some(())
 }
+
+#[derive(Default)]
+struct SignalOnDrop {
+    state: Arc<(Mutex<()>, Condvar)>,
+    thread: Option<std::thread::JoinHandle<()>>,
+}
+
+impl SignalOnDrop {
+    fn spawn_timeout(&mut self, dur: Duration, closure: impl FnOnce() + Send + 'static) {
+        let state = self.state.clone();
+        self.thread = Some(std::thread::spawn(move || {
+            let (lock, cvar) = &*state;
+            drop(cvar.wait_timeout(lock.lock().unwrap(), dur));
```

Comment:
What if there is a spurious wake up? Something that isn't related to the fuzz target completing. I think there should be an atomic or something that gets set on drop as well, and then this here should `while !atomic.load() { cvar.wait_timeout(); }`.

Reply:
It makes it a bit more complicated to handle adjustments to the timeout and such, and given the nondeterministic nature of a timeout I figured it wouldn't matter too much, but I went ahead and added in handling of spurious wakeups.